models/user/user.go

```diff
@@ -831,6 +831,21 @@ type CountUserFilter struct {
 	IsActive       optional.Option[bool]
 }
 
+// HasUsers returns true if any user exists in the database.
+// It performs a much more efficient check than counting all users.
+func HasUsers(ctx context.Context) (bool, error) {
```

Comment:
It's better to make `handleUserCreated` use the same logic.

And there is no need to add "WHERE ...", just check whether any record exists.

Reply:
There is a key logic in `handleUserCreated`, which is to determine whether the current account is the first: `CountUsers(ctx, nil) == 1`; I am not quite sure if it can be replaced with the newly added method.